app/javascript/src/NewService/components/NewServiceForm.jsx

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React, {useState, useEffect} from 'react'
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+import 'core-js/es6/promise'
+import {ServiceSourceForm, FormTemplate} from 'NewService'
+import type {SourceFormProps, FormTemplateProps} from 'NewService/types'
+
+import {createReactWrapper} from 'utilities/createReactWrapper'
+
+const BASE_PATH = '/p/admin/service_discovery'
+const NAMESPACES_URL = `${BASE_PATH}/projects.json`
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+const fetchData = async (url: string, setError) => {
+  const dataItem = url === NAMESPACES_URL
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url).then(response => {
+    if (!response.ok) {
+      throw Error(response.statusText)
+    }
+    setError({hasError: false, errorMsg: ''})
+    return response.json()
+  })
+    .then(data => data[dataItem].reduce((acc, current) => acc.concat(current.metadata.name), []))
+    .catch(error => {
+      setError({hasError: true, errorMsg: error})
+      console.error(error)
+    })
+}
+
+const getSourceFormProps = (props, handleFormsVisibility, startNamespacesFetch) => {
```

Comment:
1. Here I don't see any types for these sourceFormProps...
2. This fn might not be needed, you might want to compose a new object props out of the original props and those `handleFormsVisibility` and `startNamespacesFetch`